homeassistant/components/dyson/climate.py

```diff
@@ -333,8 +334,9 @@ def hvac_action(self):
     @property
     def fan_mode(self):
         """Return the fan setting."""
-        if self._device.state.fan_state == FanState.FAN_OFF.value:
-            return FAN_OFF
+        if self._device.state.auto_mode != AutoMode.AUTO_ON.value:
+            if self._device.state.fan_state == FanState.FAN_OFF.value:
```

Comment:
Please combine the two checks into one statement with `and`.

Reply:
I was planning to use `and` but figured that was going to go against the files own code, particularly line `295`.

Reply:
Old code can be cleaned up separately. We don't need to do that here.